File: src/codeevalchallenges/Easy/LowercasePercentageRatio_easy.py
# https://www.codeeval.com/open_challenges/147/
# Lettercase Percentage Ratio

# CHALLENGE DESCRIPTION:

# Your task is to write a program which determines (calculates) 
# the percentage ratio of lowercase and 
test = "zozkTYe"
l = len(test)
lower = 0
for letter in test:
	if letter.islower():
		lower += 1
p = lower/l
print('lowercase: %.2f uppercase: %.2f' % (p*100, (1-p)*100))

File: src/codeevalchallenges/Easy/LowercasePercentageRatio_easy.java
//https://www.codeeval.com/open_challenges/147/
//Lowercase Percentage Ratio
//
//CHALLENGE DESCRIPTION:
//Your task is to write a program which determines (calculates) the
//percentage ratio of lowercase and uppercase letters.

package codeevalchallenges.Easy;

import java.text.DecimalFormat;

class LowercasePercentageRatio {

    public static void main(String[] args) {
        String line = "thisTHIS";

        char[] stringChars = line.toCharArray();

        int lowers = 0;
        for (char x: stringChars) {
            if (Character.isLowerCase(x)) lowers++;
        }
        DecimalFormat df = new DecimalFormat("0.00");

        double lowerPerc = (double)lowers/stringChars.length *100;
        double upperPerc = (double) 100-lowerPerc;
        System.out.println("lowercase: " + df.format(lowerPerc) + " " +
                            "uppercase: " + df.format(upperPerc));
    }
}